tools/plugins/com.liferay.ide.gradle.ui/src/com/liferay/ide/gradle/ui/handler/CompareFileHandler.java

```diff
@@ -0,0 +1,234 @@
+        {
+
+            @Override
+            public void partOpened( IWorkbenchPart arg0 )
+            {
+            }
+
+            @Override
+            public void partDeactivated( IWorkbenchPart part )
+            {
+            }
+
+            @Override
+            public void partClosed( IWorkbenchPart part )
+            {
+                try
+                {
+                    IEditorReference[] editor = part.getSite().getPage().getEditorReferences();
+                    boolean isEmptyCompare = true;
+
+                    for( IEditorReference input : editor )
+                    {
+                        if( input.getId().equals( "org.eclipse.compare.CompareEditor" ) )
+                        {
+                            isEmptyCompare = !comparePages.contains( input.getEditorInput() );
+                        }
+                    }
+
+                    if( linkFolder != null && isEmptyCompare )
+                    {
+                        linkFolder.delete( true, null );
+                    }
+                }
+                catch( CoreException e )
+                {
+                    // do nothing
+                }
+            }
+
+            @Override
+            public void partBroughtToTop( IWorkbenchPart arg0 )
+            {
+            }
+
+            @Override
+            public void partActivated( IWorkbenchPart arg0 )
+            {
+            }
+        } );
+
+        ITypedElement left = null;
+        ITypedElement right = null;
+
+        left = getElementFor( getTemplateFile(currentFile) );
+        right = getElementFor( currentFile );
+
+        openInCompare( null, left, right, workBenchPage );
+    }
+
+    private void openInCompare(
+        ITypedElement ancestor, ITypedElement left, ITypedElement right, IWorkbenchPage workBenchPage )
+    {
+
+        final CompareEditorInput input = new SaveablesCompareEditorInput( ancestor, left, right, workBenchPage );
+        final IEditorPart editor =
+            Utils.findReusableCompareEditor( input, workBenchPage, new Class[] { FileRevisionEditorInput.class } );
+
+        if( editor != null )
+        {
+            IEditorInput otherInput = editor.getEditorInput();
+
+            if( otherInput.equals( input ) )
+            {
+                workBenchPage.activate( editor );
+            }
+            else
+            {
+                CompareUI.reuseCompareEditor( input, (IReusableEditor) editor );
+                workBenchPage.activate( editor );
+            }
+        }
+        else
+        {
+            CompareUI.openCompareEditor( input );
+            input.getCompareConfiguration().setLeftLabel( "Template" );;
+            input.getCompareConfiguration().setLeftEditable( false );
+            comparePages.add( input );
+        }
+    }
+
+    private ITypedElement getElementFor( IFile resource )
+    {
+        return SaveablesCompareEditorInput.createFileElement( resource );
```

Comment:
StorageTypeElement is not able to load normal java file. But we can define our TypeElement i have sent a new pull request.